tf-psa-crypto/tests/scripts/components-build-system.sh

```diff
@@ -0,0 +1,24 @@
+# components-build-system.sh
+#
+# Copyright The Mbed TLS Contributors
+# SPDX-License-Identifier: Apache-2.0 OR GPL-2.0-or-later
+
+# This file contains test components that are executed by all.sh
+
+################################################################
+#### Build System Testing
+################################################################
+
+component_test_cmake_tf_psa_crypto_out_of_source () {
```

Comment:
_Eventually_ yes, but IMO only after the transitional wrapper has been removed.

My view of the plan is as follows:
1. We have an `all.sh` in `tf-psa-crypto` but for now we don't invoke it directly on the CI, only through the transitional wrapper in `mbedtls`. (This PR + https://github.com/Mbed-TLS/TF-PSA-Crypto/issues/6 - minimal requirement for tf-psa-crypto to go live.)
2. We move all relevant components to `tf-psa-crypto` so that it gets the level of coverage we want.
3. We change the CI job on `tf-psa-crypto` to stop checking out `mbedtls` and running its `all.sh` - instead we do the normal thing of checking out `tf-psa-crypto` and running its `all.sh`. (In two steps: (a) add the new job, (b) remove the old one.)
4. We remove the transitional wrapper in `mbedtls` and we're now free to rename `tf-psa-crypto` components as we want. If we followed a consistent naming pattern, this can be as simple as `s/_tf_psa_crypto//` in `components-*.sh`.

As long as we have the transitional wrapper, ie we put `tf-psa-crypto` and `mbedtls` in the same bag, I'd really prefer names to be unique within that bag. To get rid of the naming constraints, rather than increasing the complexity associated with the transitional wrapper, let's work towards a state where we no longer need it.

Reply:
Thanks for the explanation. That looks a good plan.

Reply:
I've now included a copy of this plan in the wrapper's comments for reference.